tests/unit/moduleapi/propagate.tcl

```diff
@@ -29,14 +29,89 @@ tags "modules" {
 
                     assert_replication_stream $repl {
                         {select *}
-                        {multi}
                         {incr timer}
+                        {incr timer}
+                        {incr timer}
+                    }
+                    close_replication_stream $repl
+                }
+
+                test {module propagation with notifications} {
+                    set repl [attach_to_replication_stream]
+
+                    $master set x y
+                    wait_for_ofs_sync $master $replica
+
+                    assert_replication_stream $repl {
+                        {multi}
+                        {select *}
```

Comment:
yes, it is a bit weird but it simplifies the code tremendously.

otherwise, i would have to make propagatePendingCommands take a dbid (which i don't have, except in `call`), only to pass on to execCommandPropagateMulti which anyway does not mess with keys, so why would they need to SELECT anything?

Reply:
maybe somehow just emit the initial select in a different way?
i.e. if later someone does:
```
select 2
multi
set x x
exec
```
i'll be willing to accept that the select went into the MULTI.. but the initial one just looks very odd.
but anyway, aren't you injecting the MULTI retroactively? maybe just check that the first command is a SELECT and swap their order?
since anyway, the commands are serialized to the replication stream, which is used on the target as one client, the client state can't mix with anything so the atomicity of select doesn't matter.

unless if for some reason the transaction is aborted on the dest, in which case maybe it's a little bit better to take the select out?

Reply:
steal the dbid of MULTI from first command, EXEC from last